lib/battle/CBattleInfoCallback.cpp

```diff
@@ -1179,16 +1179,26 @@ ReachabilityInfo CBattleInfoCallback::makeBFS(const AccessibilityInfo &accessibi
 		hexq.pop();
 		
 		//walking stack can't step past the obstacles
-		//TODO what if second hex of two-hex creature enters obstacles
-		if(curHex != params.startPosition && vstd::contains(obstacles, curHex))
+		if(curHex != params.startPosition)
 		{
-			if(curHex == ESiegeHex::GATE_BRIDGE)
+			auto occupiedHexes = battle::Unit::getHexes(curHex, params.doubleWide, params.side);

```

Comment:
Lets handle correctly twohex creatures. Now they are able to go around various obstacles without hitting them with their second hex.